src/macaron/slsa_analyzer/git_service/__init__.py

```diff
@@ -1,13 +1,115 @@
-# Copyright (c) 2022 - 2022, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
 
 """The git_service package contains the supported git services for Macaron."""
 
-from .base_git_service import BaseGitService
+import os
+from configparser import ConfigParser
+
+from macaron.errors import ConfigurationError
+
+from .base_git_service import BaseGitService, NoneGitService
 from .bitbucket import BitBucket
 from .github import GitHub
 from .gitlab import GitLab
 
-# The list of supported git services. The order of the list determines the order
-# in which each git service is checked against the target repository.
-GIT_SERVICES: list[BaseGitService] = [GitHub(), GitLab(), BitBucket()]
+__all__ = [
+    "BaseGitService",
+    "BitBucket",
+    "GitHub",
+    "GitLab",
+    "NoneGitService",
+]
+
+
+def load_git_services(config: ConfigParser) -> list[BaseGitService]:
```

Comment:
From my understanding, this method and `load_git_service_domains` could be called multiple times and it will try to read the os.env value and the `.ini` config value all over again the next time we call it right? 
For example, `load_git_services` is called 1 time before the analysis start to see if there is any error. After that, it is called against in `Analyzer.get_git_service()`. Does this mean that the first one and the second one would result in different results if there are changes to the env variables we read :thinking: ?

Reply:
In the new version of my PR, environment variables are only read once at the beginning, right before the analysis starts.